OA8TV3/DownloadManager/Application/AddCommand.cs

```diff
@@ -0,0 +1,109 @@
+using DownloadManager.Infrastructure;
+
+namespace DownloadManager.Application
+{
+    internal class AddCommand : ICommand
+    {
+        public string Name => "add";
+
+        public string Description =>
+            "Új szabályt hoz létre a megadott típussal\n\tHasználat: add [szabály_típus] [feltétel] [célmappa]";
+
+        public Mode[] ValidModes => [Mode.Rules];
+        public string[] ValidArguments => ["ext", "extension", "min", "max", "pattern"];
+
+        public void Execute(IHost host, Controller controller, ICommandLoader commandLoader, string[] args)
+        {
+            if (ValidModes.Contains(controller.CurrentMode))
+            {
+                if (args.Length == 3)
+                {
+                    DestinationFolder destinationFolder;
+                    if (controller.ValidPath(args[2]))
+                    {
+                        destinationFolder = new DestinationFolder
+                        {
+                            FolderName = controller.GetFolderName(args[2]), FolderPath = args[2]
+                        };
+                    }
+                    else
+                    {
+                        string? folderPath = controller.GetPathByFolderName(args[2]);
+                        if (folderPath != null)
+                        {
+                            destinationFolder = new DestinationFolder
+                            {
+                                FolderName = args[2], FolderPath = folderPath
+                            };
+                        }
+                        else
+                        {
+                            throw new InvalidOperationException("A megadott mappa nem létezik");
+                        }
+                    }
+                    AbstractRule rule;
+                    switch (args[0])
+                    {
+                        case "ext":
+                        case "extension":
+                            {
+                                rule = new ExtensionRule { Destination = destinationFolder, Extension = args[1] };
+                                break;
+                            }
+                        case "min":
+                            {
+                                if (int.TryParse(args[1], out int value))
+                                {
+                                    rule = new SizeRule
+                                    {
+                                        Destination = destinationFolder, ComparisonType = 0, Value = value
+                                    };
+                                    break;
+                                }
+                                else
+                                {
+                                    throw new InvalidOperationException("A fájlméretnek számnak kell lennie");
+                                }
+                            }
+                        case "max":
+                            {
```

Comment:
Két dolog:
- Minden case ág lehetett volna egy külön függvény, így átláthatóbb lenne.
- A "min", "max" és egyéb argumentumokat érdemes kitenni egy konstansba és azt használni.